assets/src/blocks/Articles/ArticlesEditor.js

```diff
@@ -0,0 +1,172 @@
+import { Component, Fragment } from '@wordpress/element';
+import {
+  CheckboxControl,
+  TextControl as BaseTextControl,
+  PanelBody,
+  Tooltip
+} from '@wordpress/components';
+import { InspectorControls } from '@wordpress/block-editor';
+import withCharacterCounter from '../../components/withCharacterCounter/withCharacterCounter';
+import TagSelector from '../../components/TagSelector/TagSelector';
+import PostSelector from '../../components/PostSelector/PostSelector';
+import PostTypeSelector from '../../components/PostTypeSelector/PostTypeSelector';
+import { URLInput } from "../../components/URLInput/URLInput";
+import { ArticlesFrontend } from "./ArticlesFrontend";
+
+const { RichText } = wp.blockEditor;
+const { __ } = wp.i18n;
+
+const TextControl = withCharacterCounter(BaseTextControl);
+
+export class ArticlesEditor extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total_pages: 0
+    };
+    this.toAttribute = this.toAttribute.bind(this);
+  }
+
+  toAttribute(attributeName) {
```

Comment:
It makes sense to add this as an instance method. Just wondering, since this function will be identical in each block, if there's a way we could write it once instead of in each block?

Reply:
Another option would be to handle this in a function. That would also allow us to prevent specifying the attribute name twice. I made a branch with such a function (might need a bit of work still). https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/compare/for-attribute?expand=1

Reply:
Guess we can look at that later :+1: